equinix/resource_network_device.go

```diff
@@ -159,8 +213,14 @@ func createNetworkDeviceSchema() map[string]*schema.Schema {
 			Description: networkDeviceDescriptions["UUID"],
 		},
 		networkDeviceSchemaNames["Name"]: {
-			Type:         schema.TypeString,
-			Required:     true,
+			Type:     schema.TypeString,
+			Required: true,
+			DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
```

Comment:
Help me understand this. 

Is the remote adding `-Node0` (always 0?) to the user-supplied name? Was this a one-time change or is this applied on every user-requested name change?

Can the name be synced to what the user requested? If so, why wouldn't we update the name instead of suppressing the change?

Reply:
> Is the remote adding `-Node0` (always 0?) to the user-supplied name? Was this a one-time change or is this applied on every user-requested name change?

Since we use Node 0 device to manage the cluster, name field will always be populated with user-supplied name + "-Node0" in this case. It's a one time change for the name.  

> Can the name be synced to what the user requested? If so, why wouldn't we update the name instead of suppressing the change?

Noticed that, name is a required field here in the request. Also, each time they provision the cluster, a name change will be given for

Reply:
Ok. Per our conversation, we can follow-up with a change that will introduce:
* `name_prefix` would be added at the `equinix_network_edge_device` level, required with "cluster_details". This field will have to be set on Ready, by trimming, for the benefit of imports.
* `name` would be computed, conflicts with "name_prefix", optional, exactlyoneof "name" or "name_prefix" 

What's interesting about this is that a user can specify `name_prefix` under all conditions.

We may also want to move the `name_prefix` field into `cluster_details`. If we do that, we may want to use a different field name, like `device_name_prefix`.

Reply:
Issue opened here https://github.com/equinix/terraform-provider-equinix/issues/110